### PR TITLE
RED-60546: store redis data encoded and decode on fetching when client requires

### PR DESCRIPTION
**Pull request by amitlas**

### CODE REVIEW

File: mockredis/client.py [hunk 1]
```diff
@@ -301,11 +302,32 @@ def _rename(self, old_key, new_key, nx=False):
     def dbsize(self):
         return len(self.redis.keys())
 
-    # String Functions #
+    def _decode(self, value):
```
Comment (Amir-Hadadi) [inline mockredis/client.py:305]: All the decode=True's lead to this decode function, that doesn't necessarily decode in the end, but only decodes if decode responses is true.. so I wonder if the "decode" arguments and "_decode" are named well.. but I don't have a better name so idk..

Comment (Amir-Hadadi) [inline mockredis/client.py:305]: Having a decorator for decoding return values based on decode_responses could have been a cool solution maybe.. just a thought :) it's good as it is too

Comment (amitlas) [reply mockredis/client.py:305]: that's similar to the _encode function which(which does nothing on string), and thats the action that's done according to client request.

**Review by Amir-Hadadi — APPROVED** [review]

File: mockredis/client.py (hunk 1 above)
Comment (Amir-Hadadi) [reply mockredis/client.py:305]: Don't like that methods named decode and arguments like decode=True sometimes return stuff that aren't decoded.. maybe I would call them `use_decode_responses` but ok

**Pull request merged**